### PR TITLE
CD-44: unify templates

### PR DESCRIPTION
**Pull request by rupl**
**⚠️ Warning: I intentionally branched off of `cd-40-footer` instead of master. Merging this PR will also merge #20**

During review of #20 I noticed that some of the top-level templates had fallen out of sync with the updated to the CD assets. This PR attempts to consolidate the work as much as possible, with the admission that there's more work to be done.

Specifically, the CD footer has again been manually copied to the `_includes/footer.html` file. My attempts to link them together in a way that worked for all subdirectories were unsuccessful.

What does work well is that there's a new set of options that can be defined in front-matter under `options`. You can define an array of options. For now there are no docs aside from opening `_includes/head.html` or `_layouts/default.html` and looking at the conditionals. Some of them conflict with one another because as mentioned before I'm just reducing chaos here, not eliminating it completely. Here's a list of stuff I found:

```yml
options:
- include_cd
- include_cd_header
- include_cd_footer
- include_cd_footer_soft
- include_hid
- include_ocha_basic
- hide_header_footer

custom_css:
- any/arbitrary/path/to/css/without/extension
```

## Testing

- Run `gulp dev` and see if the CD demo still looks good
- Check footer of other pages including frontpage and ensure footer also contains logos and the like
- Check header of other pages and ensure OCHA logo is present

### CODE REVIEW

File: common-design/demo.html [hunk 1]
```diff
@@ -1,29 +1,18 @@
 ---
+options:
+- hide_header_footer
+- include_cd
+- include_cd_header
+- include_cd_footer_soft
+- include_cd_footer
```
Comment (rupl) [inline common-design/demo.html:7]: @left23 perhaps the best demonstration of what really happened here is the `demo.html` file which used to be a hardcoded collection of exceptions. Now it's running off the main layout and `<head>` markup with a series of options to change what assets/markup are loaded.

**Comment by rupl** [comment]
I rebased the PR against `cd-40-footer` so we only have to look at what I changed between your branch, not with `master` as well. The diff in #20 will grow in size once this is merged, but we'll never merge broken stuff.

File: common-design/demo.html (hunk 1 above)
Comment (left23) [reply common-design/demo.html:7]: Yes. This seems much nicer. And maybe gets us closer to including variants without having to duplicate loads of markup and files.

**Review by left23 — APPROVED** [review]
@rupl Thanks for the effort :)
Approving and merging

**Pull request merged**